Volume/Views/MainView/HomeView.swift

```diff
@@ -0,0 +1,231 @@
+//
+//  HomeView.swift
+//  Volume
+//
+//  Created by Hanzheng Li on 10/26/22.
+//  Copyright © 2022 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+
+struct HomeView: View {
+    @EnvironmentObject private var networkState: NetworkState
+    @EnvironmentObject private var userData: UserData
+    @StateObject var viewModel = ViewModel()
+
+    init() {
+        let navBarAppearance = UINavigationBarAppearance()
+        navBarAppearance.configureWithTransparentBackground()
+        navBarAppearance.backgroundColor = UIColor(Constants.backgroundColor)
+
+        UINavigationBar.appearance().standardAppearance = navBarAppearance
+        UINavigationBar.appearance().compactAppearance = navBarAppearance
+        UINavigationBar.appearance().scrollEdgeAppearance = navBarAppearance
+    }
+
+    var body: some View {
+        List {
+            Group {
+                trendingArticlesSection
+                weeklyDebriefButton
+                followedArticlesSection
+                unfollowedArticlesSection
+                deepNavigationLink
+            }
+            .listSectionSeparator(.hidden)
+            .listRowSeparator(.hidden)
+            .listRowInsets(EdgeInsets(top: 0, leading: Constants.listHorizontalPadding, bottom: 0, trailing: Constants.listHorizontalPadding))
+            .listRowBackground(Color.clear)
+        }
+        .toolbar {
+            ToolbarItem(placement: ToolbarItemPlacement.navigationBarLeading) {
+                Image.volume.logo
+                    .foregroundColor(.red)
+            }
+        }
+        .navigationBarTitleDisplayMode(.inline)
+        .listStyle(.plain)
+        .refreshable {
+            viewModel.fetchContent()
+        }
```

Comment:
```suggestion
        .refreshable(viewModel.fetchContent)
```

Reply:
also wanted to do this but it introduces warning `Converting non-sendable function value to '@Sendable () async -> Void' may introduce data races`, and I can't implement `@Sendable` on `HomeViewModel.fetchContent` because it's a `@MainActor` class.